### PR TITLE
Throw a new exception when reading empty stream.

### PR DESCRIPTION
**Pull request by LLITCHEV**
**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/EOFException.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.hedera.pbj.runtime.io.stream;
+
+import java.nio.BufferUnderflowException;
+
+/** This class is used as an exception to signal that the end of stream is reached
+ * when reading.
+ */
```
Comment (jasperpotts) [inline pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/EOFException.java:7]: minor ;-) would be nice to be one of these styles
```
/** 
  * This class is used as an exception to signal that the end of stream is reached
  * when reading.
  */
```
or
```
/**  This class is used as an exception to signal that the end of stream is reached when reading.  */
```

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java [hunk 2]
```diff
@@ -258,6 +258,25 @@ public void close() throws IOException {
...
+        final var inputStream = new ByteArrayInputStream(new byte[0]) {
+            @Override
+            public void close() throws IOException {
+                throw new IOException("Failed");
+            }
+        };
+
+        byte[] read = new byte[5];
+        final var stream = new ReadableStreamingData(inputStream);
+        try {
```
Comment (jasperpotts) [inline pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:273]: might be nice like 
```
assertThrows(EOFException.class, () -> {
      stream.readBytes(read);
});
```

**Review by jasperpotts — COMMENTED** [review]
All good just a couple minor improvements

File: pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java (hunk 2 above)
Comment (LLITCHEV) [reply pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java:273]: Changed.

File: pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/EOFException.java (hunk 1 above)
Comment (LLITCHEV) [reply pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/EOFException.java:7]: Changed the comment. It is less than 120 chars.

**Review by jasperpotts — APPROVED** [review]

**Pull request merged**